README.md

```diff
@@ -1 +1,18 @@
-# gha-tag-release
\ No newline at end of file
+# GitHub Actions - Tag release
+
+Create a tag and an optional release
+
+### Why there is no SHA input paramater
+
+Creating a tag for a particular SHA, either via the GitHub API or via CLI (i.e. git tag) in an action is strangely blocked.  The error is "Resource not accessible by integration" which is a permissions error.
```

Comment:
Have raised issue - https://support.github.com/ticket/personal/0/1657805 (probably only visible to myself)

**Ticket content:**

When trying to create a tag via a GitHub actions, called from another workflow, I can only use the most recent sha i.e. ${{ github.sha }}.

If I attempt to use a 'historic' sha e.g. one that is not sha in the current context, it fails with

"Resource not accessible by integration"

Which I understand is a permissions error, however this seems non-sensical since I can still create a tag for the current sha. I've attempted to set `permissions: write-all` in the parent workflow, though it makes no difference.

Even stranger, I CAN create a tag for a historic sha if I instead use the `https://api.github.com/repos/${{ github.repository }}/releases` end point which will create a corresponding tag if that tag does not exist

The code I tried to create the tag with is
```
# works
SHA="${{ github.sha }}"

# does not work
SHA="<historic-sha>"

curl -s \
-X POST https://api.github.com/repos/${{ github.repository }}/git/refs \
-H "Accept: application/vnd.github.v3+json" \
-H "Authorization: token ${{ github.token }}" \
-d @- << EOF
{
    "sha": "$SHA",
    "ref": "refs/tags/1.2.3"
}
EOF
```

I also get similar failure if I use CLI i.e. 
```
git checkout <historic-sha>
git tag 1.2.3
git push origin 1.2.3
```